File: taiwan-stock-frontend/src/types/scheduler.ts
export type SchedulerStatusResponse = {
    enabled: boolean;
    running: boolean;
    job_count: number;
    jobs: string[];
  };
  
  export type SchedulerJobItem = {
    id: string;
    name: string;
    next_run_time?: string | null;
    trigger: string;
    args: string[];
    paused: boolean;
  };
  
  export type SchedulerJobsResponse = {
    items: SchedulerJobItem[];
    count: number;
  };
  
  export type SchedulerControlResponse = {
    status: string;
    scheduler_running?: boolean | null;
    job_id?: string | null;
    paused?: boolean | null;
  };
  
  export type SchedulerRunNowResponse = {
    status: string;
    job_id: string;
    result: Record<string, unknown>;
  };

File: app/schemas/scheduler.py
from __future__ import annotations
from datetime import datetime

from pydantic import BaseModel


class SchedulerStatusResponse(BaseModel):
    enabled: bool
    running: bool
    job_count: int
    jobs: list[str]


class SchedulerJobItemResponse(BaseModel):
    id: str
    name: str
    next_run_time: datetime | None = None
    trigger: str
    args: list[str]
    paused: bool


class SchedulerJobsListResponse(BaseModel):
    items: list[SchedulerJobItemResponse]
    count: int


class SchedulerControlResponse(BaseModel):
    status: str
    scheduler_running: bool | None = None
    job_id: str | None = None
    paused: bool | None = None


class SchedulerRunNowResponse(BaseModel):
    status: str
    job_id: str
    result: dict

